src/ngraph/op/util/op_annotations.hpp

```diff
@@ -54,9 +54,13 @@ namespace ngraph
                     return m_in_place_oi_pairs;
                 }
 
+                bool is_cacheable() const { return m_cacheable; }
+                void set_cacheable(bool val) { m_cacheable = val; }
             private:
                 // map of output-input pairs for which in-place computation is valid
                 std::vector<struct oi_pair> m_in_place_oi_pairs;
+
+                bool m_cacheable = false;
```

Comment:
Not specific to this PR, but I really don't think we should have member variables in this class at all---including the ones that are already there. Originally, it was intended to be a completely abstract base class(though the absence of a pure virtual, or something like that, may have made that technically not the case), with specific backends subclassing it (e.g. `CPUOpAnnotations` and `GPUOpAnnotations`). I recognize that there's commonality between `CPUOpAnnotations` and `GPUOpAnnotations`, but I think we need to think a better way to handle that commonality than just jamming more and more fields into this class.

Reply:
any reason not to? I thought it was actually conceived as a kitchen sink for ad-hoc node attributes?

Reply:
I think of the `OpAnnotations` class itself as being an abstract base class for whatever particular brand of kitchen sink a backend wants. Putting a field all the way up in `OpAnnotations` means that every backend that attaches `OpAnnotations` has this field whether it wants it or not. If we think all backends need this stuff, then I don't think `OpAnnotations` is where it belongs.

Reply:
Agree with @aprocter The only issue is that when we want to use some OpAnnotation info in core passes (like what we have in MemoryLayout), we will end up with ugly dynamic_casts

Reply:
Ah, that's a tricky one! Anyway, no need to stop the presses over this.